### ISSUE TITLE
[Bug] CSP error

### ISSUE THREAD

**Issue by MatthieuF44** [tunisiano187/WME-send-to-slack#89]
[ x ] I've tryied to disable every other scripts
[ x ] I've updated the Browser
[ x ] I've updated Tampermonkey/GreaceMonkey
[ x ] I've updated the script to the latest

**Describe the bug**
When i tried to use STS script, script said that the request is send but in discord app we can't found it. After many test i observed that there is an error with the content security policy. 

In the console of developer tool, we obtain this error : 
`Refused to connect to https://discord.com/api/[REDACTED]/slack' because it violates the following Content Security Policy directive: "connect-src 'self' editor-assets.waze.com www.google-analytics.com storage.googleapis.com maps.googleapis.com www.googleapis.com stats.g.doubleclick.net clouderrorreporting.googleapis.com sheets.googleapis.com docs.google.com tigerweb.geo.census.gov services.arcgis.com"`

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: Chrome
 - Version: 104.0.5112.81
 - TamperMonkey

**Additional context**
For solving this issue temporary, I've set `Modify existing content security policy (CSP) headers` to `Remove entirely (possibly unsecure)`

**Comment by tunisiano187** [comment]
2022.08.15.01 shloud solve the problem

**Comment by MatthieuF44** [comment]
Hi,

I already test the line "@connect discord.com" but it doesn't work.

Kind regards.

**Comment by ordonezgs** [comment]
Related to CSP Policy.

The current Waze CSP Policy can contact SpreadSheets, but not Discord, Slack, Telegram and so On...

We Will have to add a `@connect` per service missing or not complying with CSP Policy.

https://github.com/tunisiano187/WME-send-to-slack/blob/cc42226417a8f68cef2d5cfe08db34e1dd627660/WME-send-to-slack.user.js#L192-L226

**Comment by ordonezgs** [comment]
Actually, reviewing it, the previous shared code it's OK.

We just have to replace the AJAX with `await MakeHTTPRequest()` and with this parameters. So we also will need to adjust the MakeHTTPRequest() to be able to make a POST request:
https://github.com/tunisiano187/WME-send-to-slack/blob/cc42226417a8f68cef2d5cfe08db34e1dd627660/WME-send-to-slack.user.js#L521-L536
and this
https://github.com/tunisiano187/WME-send-to-slack/blob/cc42226417a8f68cef2d5cfe08db34e1dd627660/WME-send-to-slack.user.js#L555-L574
even
https://github.com/tunisiano187/WME-send-to-slack/blob/cc42226417a8f68cef2d5cfe08db34e1dd627660/WME-send-to-slack.user.js#L597-L606
and finally
https://github.com/tunisiano187/WME-send-to-slack/blob/cc42226417a8f68cef2d5cfe08db34e1dd627660/WME-send-to-slack.user.js#L616-L624

This requires further dev, and implementation. Sorry, this will take time.

**Comment by ordonezgs** [comment]
Required `@connect` will be:
```JS
@connect slack.com
@connect discordapp.com
@connect discord.com
@connect telegram.org
@connect google.com
```
This will be for Google Forms, Slack, Discord and Telegram

**Comment by ordonezgs** [comment]
I have sent a request to Staff to modify their current CSP Policy to add:
```JS
discordapp.com
discord.com
api.telegram.org
hooks.slack.com
```

Let's wait for them before having to use the `await MakeHTTPRequest()` function.

**Comment by ordonezgs** [comment]
This issue is in Prod, lets wait for Staff until making the required changes.

**Comment by ordonezgs** [comment]
Staff Response:
> We are happy to let you know that ith the [WME beta release of v 2.126,](https://www.waze.com/forum/viewtopic.php?t=369711) we've released other whitelisted domains that you asked for.
Please note these are currently available in beta and will reach production once v2.126 reaches production.

**Comment by ordonezgs** [comment]
Live and working. CSP Policy returns...

```
'self'


wme-staging-frontend-assets.storage.googleapis.com


*.google-analytics.com


*.analytics.google.com


storage.googleapis.com


maps.googleapis.com


www.googleapis.com


stats.g.doubleclick.net


clouderrorreporting.googleapis.com


external.wazept.com


wazedev.com


wazedev.com:8443


sheets.googleapis.com


docs.google.com


tigerweb.geo.census.gov


services.arcgis.com


essentialintegrations.com


repl.co


jsdelivr.net


511pa.com


roadsummary.com


sfs1.roadsummary.com


deldot.gov


video.deldot.gov


ilchost.com


discordapp.com


discord.com


api.telegram.org


hooks.slack.com


limageenboite.fr


wmebookmarks.free.fr


511ny.org


511nj.org


maryland.gov


511virginia.org


newengland511.org


algotraffic.com


nvroads.com


tn.gov


511ga.org


iteriscdn.com


skyvdn.com


idrivearkansas.com


akamaihd.net


goakamai.org


trafficwise.org


ga.gov


google.com


fl511.com


511ia.org


idaho.gov


arcgis.com


kandrive.org


mass511.com


mi.us


511mn.org


modot.org


mt.gov


azureedge.net


nebraska.gov


nmroads.com


ohgo.com


tripcheck.com


ri.gov


utah.gov


ca.gov


txdot.gov


modttraffic.om


nd.gov


cttravelsmart.org


vaisala.com


skyvdn.com


cotrip.org


austintexas.gov


wyoroad.info


ncdot.gov


carsprogram.org


ksdot.org


iteris-atis.com


web-assets.waze.com;
```
